src/index.ts

```diff
@@ -1,22 +1,59 @@
-import express from "express";
+import { createServer, IncomingMessage } from "node:http";
+
+import { verifyAndParseRequest, createAckEvent } from "@copilot-extensions/preview-sdk";
 import OpenAI from "openai";
-import { verifySignatureMiddleware } from "./validate-signature.js";
+
 import { describeModel } from "./functions/describe-model.js";
 import { executeModel } from "./functions/execute-model.js";
 import { listModels } from "./functions/list-models.js";
 import { RunnerResponse } from "./functions.js";
 import { recommendModel } from "./functions/recommend-model.js";
 import { ModelsAPI } from "./models-api.js";
-const app = express();
 
-app.post("/", verifySignatureMiddleware, express.json(), async (req, res) => {
+const server = createServer(async (request, response) => {
+  if (request.method === "GET") {
+    response.statusCode = 200;
+    response.end(`OK`);
+    return;
+  }
+
+  const body = await getBody(request);
+
+  let verifyAndParseRequestResult: Awaited<ReturnType<typeof verifyAndParseRequest>>;
+  let apiKey: string;
+  try {
+    const signature = request.headers["github-public-key-signature"] as string;
+    const keyID = request.headers["github-public-key-identifier"] as string;
+    apiKey = request.headers["x-github-token"] as string;
+    verifyAndParseRequestResult = await verifyAndParseRequest(body, signature, keyID, {
+      token: apiKey,
+    });
+  } catch (err) {
+    console.error(err);
+    response.statusCode = 401
+    response.end("Unauthorized");
+    return
+  }
+
+  const { isValidRequest, payload } = verifyAndParseRequestResult
+
+  if (!isValidRequest) {
+    console.log("Signature verification failed");
+    response.statusCode = 401
+    response.end("Unauthorized");
+  }
+
+  console.log("Signature verified");
+
   // Use the GitHub API token sent in the request
-  const apiKey = req.get("X-GitHub-Token");
   if (!apiKey) {
-    res.status(400).end();
+    response.statusCode = 400
+    response.end()
     return;
   }
 
+  response.write(createAckEvent().toString());
```

Comment:
I don't love the idea of this method - it feels unnecessary, and "Ack" isn't a super common term. The functionality of the integration doesn't change at all without calling this.

Separately: can we make the `create*Event` methods just return a string? Is there a case where you need the object returned, and not just the string?

Reply:
> The functionality of the integration doesn't change at all without calling this.

it does change in the chat UI, it shows that the agent is responding. But yeah it's optional, feel free to remove it if you don't like it, but I would consider it best practice.

> Separately: can we make the `create*Event` methods just return a string? Is there a case where you need the object returned, and not just the string?

These are super low-level methods. I was hoping that `response.write` would call the `.toString()` method directly, but alas, it doesn't. I think it makes sense for them to create an object in case you want to use your own logic to transport them. Once they are a string, the structural data can no longer be retrieved. 

The higher-level API would remove make the problem of calling `.toString()` go away though:
https://github.com/copilot-extensions/preview-sdk.js/blob/9533a1fd20409332c8f1cf60ac15e2421347ce55/dreamcode.md#L35-L43

Reply:
you know what I changed my mind, let's just have them generate strings, it's a problem when it's a problem. I'll apply the change in #6 if that's okay with you, let's get that one merged first, ok?

Reply:
> and "Ack" isn't a super common term

yeah naming is hard 🤔 What would you call it? Maybe avoid the abbreviation `createAcknowledgeEvent`? Or call it `createConfirmRequestEvent()`? The latter is too close to [`createConfirmationEvent()`](https://github.com/copilot-extensions/preview-sdk.js?tab=readme-ov-file#createconfirmationevent-id-title-message-metadata-) for my taste though

Reply:
it's done: https://github.com/copilot-extensions/preview-sdk.js/releases/tag/v4.0.0

Reply:
I'll go with `createAcknowledgementEvent()` for now, more coherent than `createAcknowledgeEvent()` with the other `create*Event()` methods

Reply:
> it does put the chat UI into "the agent received your request and is responding" mode, the UI changes. I would consider it best practice. But we can remove it if you prefer

This points to an improvement we should make to the platform; _we_ already know when we've started communicating to an external agent. I'd prefer to remove that for now if that's cool!

Reply:
I removed it via [7d5dd21](https://github.com/copilot-extensions/github-models-extension/pull/5/commits/7d5dd210f45960513f14b926c22d896ba17bcb70)